src/arch/riscv/isa/templates/vector_mem.isa

```diff
@@ -523,9 +544,11 @@ Fault
     }
     %(op_decl)s;
     %(op_rd)s;
+    %(set_vlenb)s;
     %(ea_code)s;
 
-    for (size_t i = 0; i < VLENB; i++) {
+
+    for (size_t i = 0; i < vlenb; i++) {
         %(memacc_code)s;
     }
 
```

Comment:
Please specify the size of memory for write, the method will read the memory data with maximum vector length size.